opt_einsum/tests/test_parser.py

```diff
@@ -24,6 +24,19 @@ def test_parse_einsum_input():
     assert operands == ops
 
 
+def test_parse_einsum_input_shapes_error():
+    eq = "ab,bc,cd"
+    ops = [np.random.rand(2, 3), np.random.rand(3, 4), np.random.rand(4, 5)]
+
+    try:
```

Comment:
For pytest you can use `with pytest.raises(ValueError)` here: https://docs.pytest.org/en/7.1.x/how-to/assert.html

Reply:
Ah, I just avoided pulling in dependencies.